messaging/src/main/java/org/axonframework/commandhandling/gateway/AbstractRetryScheduler.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2010-2019. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.AxonNonTransientException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+import static org.axonframework.common.BuilderUtils.assertStrictPositive;
+
+/**
+ * An abstract base class for {@link RetryScheduler}s. This class provides methods to do the actual rescheduling and
+ * decides if a given {@link Throwable} is explicitly transient.
+ *
+ * @author Bert Laverman
+ * @since 4.2
+ */
+public abstract class AbstractRetryScheduler implements RetryScheduler {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private static final int DEFAULT_MAX_RETRIES = 1;
+
+    private final ScheduledExecutorService retryExecutor;
+    private final int maxRetryCount;
+
+    protected AbstractRetryScheduler(Builder builder) {
```

Comment:
Nit: I think it would be nice to give this constructor some javadoc.

Reply:
Done

Reply:
Thanks!